sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -1323,4 +1323,80 @@ unsafe fn create_written_object<'vm, 'state>(
     )
 }
 
+impl<'vm, 'state, 'a> DataStore for ExecutionContext<'vm, 'state, 'a> {
+    fn link_context(&self) -> AccountAddress {
+        self.linkage_view.link_context()
+    }
+
+    fn relocate(&self, module_id: &ModuleId) -> PartialVMResult<ModuleId> {
+        self.linkage_view.relocate(module_id).map_err(|err| {
+            PartialVMError::new(StatusCode::LINKER_ERROR)
+                .with_message(format!("Error relocating {module_id}: {err:?}"))
+        })
+    }
+
+    fn defining_module(
+        &self,
+        runtime_id: &ModuleId,
+        struct_: &IdentStr,
+    ) -> PartialVMResult<ModuleId> {
+        self.linkage_view.defining_module(runtime_id, struct_).map_err(|err| {
+            PartialVMError::new(StatusCode::LINKER_ERROR).with_message(format!(
+                "Error finding defining module for {runtime_id}::{struct_}: {err:?}"
+            ))
+        })
+    }
+
+    fn load_module(&self, module_id: &ModuleId) -> VMResult<Vec<u8>> {
+        if let Some(bytes) = self.get_module(module_id) {
+            return Ok(bytes.clone());
+        }
+        match self.state_view.get_module(module_id) {
+            Ok(Some(bytes)) => Ok(bytes),
+            Ok(None) => Err(PartialVMError::new(StatusCode::LINKER_ERROR)
+                .with_message(format!("Cannot find {:?} in data cache", module_id))
+                .finish(Location::Undefined)),
+            Err(err) => {
+                let msg = format!("Unexpected storage error: {:?}", err);
+                Err(
+                    PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                        .with_message(msg)
+                        .finish(Location::Undefined),
+                )
+            }
+        }
+    }
+
+    fn publish_module(&mut self, _module_id: &ModuleId, _blob: Vec<u8>) -> VMResult<()> {
```

Comment:
no we cannot panic here because the VM/runtime calls this. At a later point in time we can (and most likely will) change the traits and callback used for the `DataStore`, but the way things are now that is a call we just have to do nothing, or we hook up instead of the work we do in publishing